jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,811 @@
+            new DumpableCollection("excluded paths", _pathSpecs.getExcluded()),
+            new DumpableCollection("excluded predicates", _predicates.getExcluded())
+        );
+    }
+
+    /**
+     * Create a {@link Predicate} over {@link Request} built from the {@link Predicate#and(Predicate) and} of one or more of: <ul>
+     *     <li>{@link ConnectorPredicate}</li>
+     *     <li>{@link InetAddressPatternPredicate}</li>
+     *     <li>{@link MethodPredicate}</li>
+     *     <li>{@link PathSpecPredicate}</li>
+     * </ul>
+     * @param connectorName The connector name or {@code null}
+     * @param inetAddressPattern An {@link InetAddressPattern} string or {@code null}
+     * @param method A {@link org.eclipse.jetty.http.HttpMethod} name or {@code null}
+     * @param pathSpec A {@link PathSpec} string or {@code null}
+     * @return the combined {@link Predicate} over {@link Request}
+     */
+    public static Predicate<Request> from(String connectorName, String inetAddressPattern, String method, String pathSpec)
+    {
+        return from(connectorName, InetAddressPattern.from(inetAddressPattern), method, pathSpec == null ? null : PathSpec.from(pathSpec));
+    }
+
+    /**
+     * Create a {@link Predicate} over {@link Request} built from the {@link Predicate#and(Predicate) and} of one or more of: <ul>
+     *     <li>{@link TypeUtil#truePredicate()}</li>
+     *     <li>{@link ConnectorPredicate}</li>
+     *     <li>{@link InetAddressPatternPredicate}</li>
+     *     <li>{@link MethodPredicate}</li>
+     *     <li>{@link PathSpecPredicate}</li>
+     * </ul>
+     * @param connectorName The connector name or {@code null}
+     * @param inetAddressPattern An {@link InetAddressPattern} or {@code null}
+     * @param method A {@link org.eclipse.jetty.http.HttpMethod} name or {@code null}
+     * @param pathSpec A {@link PathSpec} or {@code null}
+     * @return the combined {@link Predicate} over {@link Request}
+     */
+    public static Predicate<Request> from(String connectorName, InetAddressPattern inetAddressPattern, String method, PathSpec pathSpec)
+    {
+        Predicate<Request> predicate = TypeUtil.truePredicate();
+
+        if (connectorName != null)
+            predicate = predicate.and(new ConnectorPredicate(connectorName));
+
+        if (inetAddressPattern != null)
+            predicate = predicate.and(new InetAddressPatternPredicate(inetAddressPattern));
+
+        if (method != null)
+            predicate = predicate.and(new MethodPredicate(method));
+
+        if (pathSpec != null)
+            predicate = predicate.and(new PathSpecPredicate(pathSpec));
+        
+        return predicate;
+    }
+
+    /**
+     * A Set of {@link Predicate} over {@link Request} optimized for use by {@link IncludeExclude}.
+     */
+    public static class PredicateSet extends AbstractSet<Predicate<Request>> implements Set<Predicate<Request>>, Predicate<Request>
```

Comment:
This class is named as a `Set` but behaves like a `List`.

Reply:
It behaves as a `Set` but is implemented with a `List`.
I've fixed add so it does not allow duplicates.